backend/State.mo

```diff
@@ -228,8 +230,7 @@ module {
     let equal = (Text.equal, Text.equal);
     let hash = (Text.hash, Text.hash);
     func messageEqual(a: Types.Message, b: Types.Message) : Bool = a == b;
-    // not a very good hash, but we are not using the hash
-    func messageHash(m: Types.Message) : Hash.Hash = Int.hash(m.time);
+    func messageHash(m: Types.Message) : Hash.Hash = Int.hash(m.id); // id is unique, so hash is unique
```

Comment:
@osa1 I hope that this will overcome the assertion failure that you saw in your tests